Story/OnlineUIComponents/SpectatorHud.cs

```diff
@@ -46,23 +46,9 @@ public override void Update()
             base.Update();
             if (spectatorOverlay != null)
             {
-                if (RainMeadow.isStoryMode(out var _))
+                if (RainMeadow.isStoryMode(out _) || RainMeadow.isArenaMode(out _))
                 {
-                    if ((game.pauseMenu != null || camera.hud.map.visible || game.manager.upcomingProcess != null))
-                    {
-                        RainMeadow.Debug("Shutting down spectator overlay due to another process request");
-                        spectatorOverlay.ShutDownProcess();
-                        spectatorOverlay = null;
-                        return;
-                    }
-
-
-                }
-
-                if (RainMeadow.isArenaMode(out var _))
-                {
-
-                    if (game.arenaOverlay != null || game.pauseMenu != null || game.manager.upcomingProcess != null)
+                    if (game.pauseMenu != null || game.arenaOverlay != null || camera.hud.map.visible || game.manager.upcomingProcess != null)
```

Comment:
heads up, camera hud map check will break arena

Reply:
changed it to `camera.hud?.map?.visible is true`